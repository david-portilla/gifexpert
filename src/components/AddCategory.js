import React, { useState } from 'react'

export const AddCategory = () => {

  const [ inputValue, setInputValue ] = useState('Hello world')

  const handleInputChange = (e) => {
    if (e) setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    // console.log(e)
  }

  return (
    <form onSubmit={ handleSubmit }>
      <input
        type="text"
        value={ inputValue }
        onChange={ handleInputChange }
        onClick={ () => { setInputValue('') } }
      />
    </form>
  )
}

