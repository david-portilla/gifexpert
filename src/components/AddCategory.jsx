import React, { useState } from "react";

export const AddCategory = ({ onSearchTerm }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 2) {
      console.log("please writte a term");
      return;
    }
    setInputValue("");
    onSearchTerm(inputValue.trim());
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="Search gif ..."
          value={inputValue}
          onChange={handleChange}
        />
      </form>
      <button onClick={handleSubmit}>Search</button>
    </>
  );
};
