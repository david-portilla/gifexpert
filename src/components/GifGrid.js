import React from 'react'

export const GifGrid = ({ category }) => {

  const getImages = (async () => {
    const URL = 'https://api.giphy.com/v1/gifs/search?q=vegeta&limit=10&api_key=oC8I9foQB9Y1uSSK4dEKfF38kcVQFAEa'

    const res = await fetch(URL)
    const { data } = await res.json()

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        ulr: img.images?.downsized_medium.url
      }
    })
    console.log(gifs)
  })()

  return (
    <>
      <h3 key={ category }>{ category }</h3>
    </>
  )
}
