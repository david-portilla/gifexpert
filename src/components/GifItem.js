import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img key={ id } src={ url } alt={ title } />
      <p> { title }</p>
    </div>
  )
}

GifItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

export default GifItem
