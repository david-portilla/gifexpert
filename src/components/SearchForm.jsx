import React, { useState } from "react";

export const SearchForm = ({ onSearchTerm }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 3) return;
    setInputValue("");
    onSearchTerm(inputValue.trim());
  };

  return (
    <>
      {inputValue.trim().length <= 3 && <p>Please write a search term.</p>}
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="Search gif ..."
          value={inputValue}
          onChange={handleChange}
        />
      </form>
      <button onClick={handleSubmit} disabled={inputValue.trim().length <= 3}>
        Search
      </button>
    </>
  );
};
