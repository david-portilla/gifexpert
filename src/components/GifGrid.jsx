import React from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ term }) => {
  const { images, isLoading } = useFetchGifs(term);

  return (
    <>
      {term.length > 0 && images.length === 0 && (
        <p>No results were found for:</p>
      )}
      <h3>{term}</h3>
      {isLoading && <h2>Loading</h2>}
      <div className="card-grid">
        {images &&
          images.map(({ id, title, url }) => (
            <GifItem key={id} title={title} url={url} />
          ))}
      </div>
    </>
  );
};
