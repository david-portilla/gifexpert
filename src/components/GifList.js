import React from 'react'
import GifItem from './GifItem'
import { useFetchGifs } from './hooks/useFecthGifs'

export const GifList = ({ category }) => {

  const { data:images, loading } = useFetchGifs(category)

  return (
    <div className="card-list">
      <h3 className="card-list-title" key={ category }>{ category }</h3>

      { loading && <p>cargando</p> }

      {
        images.map(img => (
          <GifItem
            key={ img.id }
            { ...img } />
        ))
      }
    </div>
  )
}
