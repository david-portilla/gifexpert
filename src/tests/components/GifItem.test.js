import React from 'react'
import { shallow } from 'enzyme';
import { GifItem } from '../../components/GifItem';

describe('Testing <GifItem />', () => {
  const id = 'dadad'
  const title = 'gif title'
  const url = 'www.testingmywebsite.com'

  test('should show <GifItem />', () => {
    const wrapper = shallow(<GifItem id={ id } title={ title } url={ url } />)
    expect(wrapper).toMatchSnapshot()
  })
})