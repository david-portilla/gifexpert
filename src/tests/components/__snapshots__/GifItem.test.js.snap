// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> should show <GifItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="gif title"
    key="dadad"
    src="www.testingmywebsite.com"
  />
  <p>
     
    gif title
  </p>
</div>
`;
